### PR TITLE
notification components

### PR DESCRIPTION
**Pull request by IgorOsiyuk**
create and set up notifications for keep's request
image
image

There are notifications for successful requests when we update, delete and post keeps
We also catch bad requests and show notification for this error in red

login: user@user.com
password: 1234567890

### CODE REVIEW

File: google_keeps/src/components/Notification/index.jsx [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+  });
+};
+
+export const updateNotification = () => {
+  store.addNotification({
+    ...notificationConfig,
+    message: "Your keep was updated",
+    type: "info",
+  });
+};
```
Comment (melonholy) [inline google_keeps/src/components/Notification/index.jsx:32]: maybe you should name it like infoNotification and pass message as argument

File: google_keeps/src/components/Notification/index.jsx [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+export const newKeepNotification = () => {
+  store.addNotification({
+    ...notificationConfig,
+    message: "Your keep was created",
+    type: "success",
+  });
+};
```
Comment (melonholy) [inline google_keeps/src/components/Notification/index.jsx:48]: name it like successNotification and pass message as argument

File: google_keeps/src/components/keeps/NewKeepForm.jsx [hunk 3]
```diff
@@ -135,8 +135,8 @@ const NewKeepForm = () => {
   }, [isEmptyField]);
 
   const sendNewKeep = useCallback(() => {
-    dispatch(PostUserKeepRequest(keepData)).then(() => {
-      setKeepData(defaultState);
+    dispatch(PostUserKeepRequest(keepData)).then((res) => {
+      res && setKeepData(defaultState);
```
Comment (melonholy) [inline google_keeps/src/components/keeps/NewKeepForm.jsx:139]: hmmm is it required check?

File: google_keeps/src/components/pages/Keeps.jsx [hunk 4]
```diff
@@ -5,20 +5,32 @@ import { Container } from "../shared/base";
...
-    <Container>
-      <KeepCreate/>
-      <KeepsList list={listOfKeeps} />
-    </Container>
+    <>
+      <Container>
+        <KeepCreate />
+        <KeepsList list={listOfKeeps} />
+      </Container>
+      <Notification />
```
Comment (melonholy) [inline google_keeps/src/components/pages/Keeps.jsx:32]: I suggest to move notification to the higher level of app

File: google_keeps/src/redux/reducers/user.jsx [hunk 5]
```diff
@@ -1,7 +1,11 @@
 import * as actionType from "../action/types";
 
 const initialState = {
-  keeps:[]
+  keeps: [],
+  error: {
+    isError: false,
+    errorList: null,
```
Comment (melonholy) [inline google_keeps/src/redux/reducers/user.jsx:7]: probably errorList should be empty arr by default

**Review by melonholy — CHANGES_REQUESTED** [review]

File: google_keeps/src/components/Notification/index.jsx (hunk 1 above)
Comment (IgorOsiyuk) [reply google_keeps/src/components/Notification/index.jsx:32]: - [+]

File: google_keeps/src/components/Notification/index.jsx (hunk 2 above)
Comment (IgorOsiyuk) [reply google_keeps/src/components/Notification/index.jsx:48]: - [+]

File: google_keeps/src/redux/reducers/user.jsx (hunk 5 above)
Comment (IgorOsiyuk) [reply google_keeps/src/redux/reducers/user.jsx:7]: - [+]

File: google_keeps/src/components/pages/Keeps.jsx (hunk 4 above)
Comment (IgorOsiyuk) [reply google_keeps/src/components/pages/Keeps.jsx:32]: - [+] 
- I moved it to App component

File: google_keeps/src/components/keeps/NewKeepForm.jsx (hunk 3 above)
Comment (IgorOsiyuk) [reply google_keeps/src/components/keeps/NewKeepForm.jsx:139]: that allows me to understand that post-request completed successfully and after that I clear all fields in my NewKeep form

**Review by melonholy — APPROVED** [review]

**Pull request merged**